kem/tests/hpke.rs

```diff
@@ -21,7 +21,17 @@ impl Encapsulate<EncappedKey, SharedSecret> for PublicKey {
         &self,
         mut csprng: &mut impl CryptoRngCore,
     ) -> Result<(EncappedKey, SharedSecret), HpkeError> {
-        <X25519HkdfSha256 as KemTrait>::encap(&self.0, None, &mut csprng).map(|(ek, ss)| (ss, ek))
+        <X25519HkdfSha256 as KemTrait>::encap(&self.0, None, &mut csprng).map(|(ss, ek)| (ek, ss))
+    }
+
+    fn encapsulate_in_place(
+        &self,
+        csprng: &mut impl CryptoRngCore,
+        encapsulated_key: &mut EncappedKey,
+    ) -> Result<SharedSecret, HpkeError> {
+        let (ek, ss) = self.encapsulate(csprng)?;
+        *encapsulated_key = ek;
+        Ok(ss)
     }
```

Comment:
Could this be the default impl?

Reply:
Ya that can work. If `EK: Default` then the other method could be the default impl. Which is to say there’s no “natural” lower-level impl here. But ya maybe that doesn’t matter and we just make one default bc it makes life easier for the implementers

Reply:
I think it's more common for in-place operations to be provided so you only have to write them in cases where optimizations are possible. See e.g. `Clone::clone` vs `Clone::clone_from`

Reply:
Added a default impl. Thank you for the good idea